modules/metastorage-client/src/main/java/org/apache/ignite/internal/metastorage/client/MetaStorageServiceImpl.java

```diff
@@ -533,6 +538,7 @@ private final class Watcher extends Thread {
              * @param lsnr   The listener which receives and handles watch updates.
              */
             Watcher(Cursor<WatchEvent> cursor, WatchListener lsnr) {
+                setName("ms-watcher-" + localNodeName);
```

Comment:
It would be better to use `IgniteThread` instead of `Watcher extends Thread` I think. Will file a ticket for this.

Reply:
I agree, we don't need to extend java's basic threads when we have ignite thread